endtoend/evaluators/data.go

```diff
@@ -0,0 +1,44 @@
+package evaluators
+
+import (
+	"context"
+	"errors"
+
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/endtoend/types"
+	"google.golang.org/grpc"
+)
+
+const epochToCheck = 50 // must be more than 46 (32 hot states + 16 chkpt interval)
+
+// ColdStateCheckpoint checks data from the database using cold state storage.
+var ColdStateCheckpoint = types.Evaluator{
+	Name: "cold_state_assignments_from_epoch_%d",
+	Policy: func(currentEpoch uint64) bool {
+		return currentEpoch == epochToCheck
+	},
+	Evaluation: checkColdStateCheckpoint,
+}
+
+// Checks the first node for an old checkpoint using cold state storage.
+func checkColdStateCheckpoint(conns ...*grpc.ClientConn) error {
+	ctx := context.Background()
+	client := eth.NewBeaconChainClient(conns[0])
+
+	for i := uint64(0); i < epochToCheck; i++ {
+		// This request for epoch 24 should use the cold state checkpoint from epoch 16.
```

Comment:
This comment seems inaccurate now. Also, should we request this 50 times?

Reply:
Oops, I thought I moved this comment, but seems I only copied it.

```suggestion
```

Yes, @terencechain's feedback was that we should check several epochs. Checking every epoch since genesis should give us good coverage of this feature and we only do it once so it should be OK to do.